src/Components/common/Auth0LoginButton/Auth0LoginButton.js

```diff
@@ -0,0 +1,101 @@
+import React, { useContext } from "react";
+import { store as loginStore } from "../../Login/LoginContainer";
+import { store as registerStore } from "../../Register/RegisterContainer";
+import { HANDLE_SOCIAL_LOGIN } from "../../../utils/action-types";
+import { ReactComponent as Auth0LogoIcon } from "../../../assets/auth0-logo.svg";
+
+export const Auth0LoginButton = ({
+  label = "Auth0",
+  className = "",
+  labelClassName = "",
+  iconClassName = ""
+}) => {
+  const auth0Enabled = Boolean(
+    window.Pelcro.site.read().auth0_client_id &&
+      window.Pelcro.site.read().auth0_base_url
+  );
+
+  const auth0InstanceRef = React.useRef(null);
+  React.useEffect(() => {
+    if (auth0Enabled) {
+      auth0InstanceRef.current = new window.auth0.WebAuth({
+        domain: window.Pelcro.site.read().auth0_base_url,
+        clientID: window.Pelcro.site.read().auth0_client_id
+      });
+    }
+  }, []);
+
+  const { dispatch: loginDispatch } = useContext(loginStore);
+  const { dispatch: registerDispatch } = useContext(registerStore);
+
+  function handleClick() {
+    auth0InstanceRef.current?.popup?.authorize?.(
+      {
+        responseType: "token id_token",
+        owp: true // close popup when finished
+      },
+      function (error, authResult) {
+        if (error) {
+          return onFailure(error);
+        }
+
+        onSuccess(authResult);
+      }
+    );
+  }
+
+  const onSuccess = (authResult) => {
```

Comment:
we usually have this kind of logic as part of the container, why did we include it here in this case?

Reply:
What I'm doing is dispatching the login success event to the container to handle the auth logic. what the button is doing in onSuccess is basically extracting the user data from the auth0 sdk.